src/notes/note.utils.ts

```diff
@@ -1,18 +1,53 @@
-import { Note } from './note.entity';
+import { safeLoad } from 'js-yaml'
 
 export class NoteUtils {
-  public static parseTitle(note: Note): string {
-    // TODO: Implement method
-    return 'Hardcoded note title';
+  public static parseTitle(content: string): string {
+    const metadata = NoteUtils.parseMetadata(content);
+    if (metadata['title']) {
+      return metadata['title']
+    }
+
+    if (metadata['opengraph'] && metadata['opengraph']['title']) {
+      return metadata['opengraph']['title']
+    }
+
+    try {
+      const firstHeading = content.indexOf('# ')
+      const endOfHeading = content.indexOf('\n', firstHeading)
+      return content.substr(firstHeading, endOfHeading - firstHeading).substr(2)
+    } catch (err) {
+      return ''
+    }
+  }
+
+  public static parseDescription(content: string): string {
+    const metadata = NoteUtils.parseMetadata(content);
+    if (metadata['description']) {
+      return metadata['description']
+    }
+
+    if (metadata['opengraph'] && metadata['opengraph']['description']) {
+      return metadata['opengraph']['description']
+    }
+
+    return ''
   }
 
-  public static parseDescription(note: Note): string {
-    // TODO: Implement method
-    return 'Hardcoded note description';
+  public static parseTags(content: string): string[] {
+    const metadata = NoteUtils.parseMetadata(content);
+
+    if (metadata['tags']) {
+      return metadata['tags'].split(',').map(entry => entry.trim()) ?? []
+    }
+
+    return []
   }
 
-  public static parseTags(note: Note): string[] {
-    // TODO: Implement method
-    return ['Hardcoded note tag'];
+  private static parseMetadata(content: string) {
+    const matches = content.match(/^---\n((?:.|\n)*)\n---.*/)
+    if (!matches || matches.length <= 1) {
+      return ''
+    }
+    return safeLoad(matches[1])
```

Comment:
Do we need this functions for other things than parsing from note? Or why did you change the signature?

Reply:
Because it would be much more cumbersome to fetch the current text from the `Note` object than just get it via the parameters. We would need to add the `getLastRevision()` function to `NoteUtils` or the `Note` object itself…